runtime/compiler/env/J9ClassEnv.hpp

```diff
@@ -121,6 +121,21 @@ class OMR_EXTENSIBLE ClassEnv : public OMR::ClassEnvConnector
     */
    bool classHasIdentity(TR_OpaqueClassBlock *clazz);
 
+   /** \brief
+    *    Checks whether a class contains any field of type double (D), float (F),
+    *    nullable-class/interface type (L) or null-free class type (Q) that are
+    *    not both flattened and recursively compatible for the fast substitutability
+    *    optimization.
+    *
+    *  \param clazz
+    *     The class that is to be checked
+    *
+    *  \return
+    *    `true` if the class is compatible for the fast substitutability optimization;
+    *    `false` otherwise.
+    */
+   bool classCanSupportFastSubstitutability(TR_OpaqueClassBlock *clazz);
```

Comment:
I realize that this name comes from the name of the flag used in the JVM, but I wonder whether we could use a name that's more meaningful - maybe something like `classSupportsBitwiseComparison` or `classSupportsDirectMemoryComparison`.

Reply:
Addressed in 307194d